ArduPilot/ardupilot
# Airspeed: Add CAN node overides

This PR adds the abbility to force specific CAN node IDs to certain instances of airspeed.  This is useful in aplications where we care where each airspeed sensor is on the aircraft, whether is should be used for logging only or active flight measurement.  As a very crude example.... my high accuracy cardboard AoA sensor that I was messing around with.  We care about which airspeed sensor is which for the AoA calculation and we don't want to to use the -15 deg pitch sensor for flight.
![ais_AoA](https://user-images.githubusercontent.com/34512430/162611797-adc8566b-4f13-49fc-8a97-6301d7f5519b.jpg)

This PR does depend on the refactor I have done in this PR: #20497 , splitting them out to make review a little easier/incremental.  The key thing in this PR is the last commit: [AP_Airspeed: Add CAN node id overide](https://github.com/ArduPilot/ardupilot/commit/a2122eba32dd31b2cb8554c7fee82243c27a029b)

Done a fair bit of testing on this using my airspeed sensor pan pipes 😃 
![aispanpipes](https://user-images.githubusercontent.com/34512430/162611822-0dfb4146-c746-4305-ab34-9b91e99be931.jpg)

Tests done:
- Set all instances at different overrides and makes sure they repeatedly get assigned to the same instance
- Set the same overide on two instances.  In which case the first matching instance is allowed to init, and the second clash is not.  Here, there is a GCS message to warn the user, but i am fairly sure GCS is not init at this point and so the message does not get through.  The saving grace however is that the sensor is marked unhealthy so generates an arming warning in plane.
- Set a mix of overriden and non-overiden and make sure that they all init correctly, and that the non-overidden instance does not 'steal' a node that is overidden in another instance.
